lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1402,3 +1424,70 @@ def _get_deployable_vhd_path(
             wait_copy_blob(blob_client, vhd_path, log)
 
         return full_vhd_path
+
+    def _node_generate_min_capability(
+        self, req_cap: schema.NodeSpace, azure_cap: schema.NodeSpace
+    ) -> schema.NodeSpace:
+        assert azure_cap.features, "azure node has no disk features."
+        assert (
+            AzurePlatform._get_disk_feature_count(req_cap.features) <= 1
+        ), "Multiple disk feature found in node requirement."
+
+        if AzurePlatform._get_disk_feature_count(req_cap.features) == 0:
+            # If no disk type feature is specified, use least
+            # cost available disk from azure_cap
+            min_cost_disk = AzurePlatform._get_min_cost_disk(azure_cap.features)
+        else:
+            min_cost_disk = AzurePlatform._get_min_cost_disk(
+                req_cap.features  # type: ignore
+            )
+
+        # Remove mutually exclusive feature
+        azure_cap_copy = copy.deepcopy(azure_cap)
+        azure_cap_copy.features.update(  # type: ignore
+            copy.deepcopy(azure_cap.features.items)
+        )
+        for me_feature_group in features.MUTUALLY_EXCLUSIVE_FEATURES:
+            if min_cost_disk in me_feature_group:
+                features_to_remove = set(me_feature_group)
+                features_to_remove.discard(min_cost_disk)
+                azure_cap_copy.features.difference_update(  # type: ignore
```

Comment:
use assert to avoid this kind of type: ignore

Reply:
This assert would be redundant. This is deepcopy of `azure_cap` and we already have a assert on it. I feel it makes the logic more complex. Please let me know if you think otherwise.

Reply:
The assert is better than "# type: ignore". It's to improve typing validation.

Reply:
There is another `#type : ignore` present on https://github.com/microsoft/lisa/blob/8f703e8511be0a77f7fbf6a1cf51d58eb612dbe9/lisa/sut_orchestrator/azure/platform_.py#L1436-L1443

If counter is in else block, `req_cap.features` is greater than 0. But mypy is still throwing an error. I am thinking of adding a `None` check to `if` condition rather than an `assert` in else block.

Reply:
mypy is a static check tool, it's not smart enough to understand this kind of logic. Yes, you can add a None check in the if block.